ledger-service/http-json/src/main/scala/com/digitalasset/http/SchemaHandlingResult.scala

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http
+
+import com.daml.http.dbbackend.ContractDao
+import com.daml.logging.{ContextualizedLogger, LoggingContextOf}
+import cats.effect.IO
+import com.daml.http.util.Logging.InstanceUUID
+
+sealed trait SchemaHandlingResult
+object SchemaHandlingResult {
+  case object Terminate extends SchemaHandlingResult
```

Comment:
The logic here seems a bit awkward. I would suggest something like the following:
1. Have a flag for whether you want to start the server after schema initialization. This can either be a separate flag in `Config` or a `shouldStart` method on SchemaHandling. Handle this at the call site not inside `fromSchemaHandling`.
2. Rename `fromSchemaHandling` to something like `createAndValidateSchema` and depending on the flag only validate, always recreate or recreate if needed.
3. `createAndValidateSchema` either succeeds, in that case continue checking `shouldStart` and start if it’s true, otherwise exit with code 0. If createAndValidateSchema fails, always terminate with a non-zero exit code.

Reply:
@realvictorprm you resolved this comment but you didn’t address it fully. You still conflate errors during initialization (e.g. version mismatch) with the user selecting to exit after creating the schema and both turn into exit code 0.

My suggestion above was to only handle "error vs no error" here and check if you continue and the call site rather than in here.

Reply:
True, I did not respect the error codes. They were not as important in my mind yet, but I see why it is important to keep these right. Will do a fancy fix.

Reply:
I think this should be OK now, the exit codes are now fixed.